### DISCUSSION TITLE
Add default callback to `fx.prepend` method

### DISCUSSION THREAD

**Discussion by sergeysova** [zerobias/effector#432] [Archive]
## Motivation

Sometimes I need to call effect from my React-components. WebStorm shows me a warning about "unhandled promise" (`Promise returned from myFx is ignored`), because I call effect in event handler, without `await`. IDE assumes that promise can be rejected, and in Node.JS environment  it can be "unhandled promise rejection" and [terminate process](https://github.com/nodejs/node/issues/20392), but IDE don't know about [that fix in effect source code](https://github.com/effector/effector/commit/a00251b67f97911c1c83869263529890e6931b0c).

## Current solution

I creates a new event from effect using `.prepend` with noop callback or identity-callback, but typescript don't like identity-callback if effect has no arguments, also `prepend` requires to set type for callback argument, that annoying.

```ts
const foo = fxWithoutArgs.prepend(() => {});
// and
const bar = generalFx.prepend(args => args);
```

## Suggestion

I suggest to add default callback `arg => arg` and add types for typescript, to allow `void` params.

```ts
const foo: Event<void> = fxWithoutArgs.prepend();
// and
const bar: Event<Params> = generalFx.prepend();
```

**Comment by zerobias** [comment]
As we artificially forced promise returned by effect to be always handled, I'm suggest to ask jetbrains to allow exceptions for their linting, because changing runtime and introducing additional complexity to just cool down a thing builded to help us is not an optimal way to fix issues

**Comment by doasync** [comment]
I just write:
`
void getDataFx()
`
according to my linter rules in order to indicate that promise is ignored intentionally
